### ISSUE TITLE
RDP is disabled after 30 seconds / kicked out of RDP session

### ISSUE THREAD

**Issue by asheroto** [daleffe/rdp-port-knocking#4]
Hello there!

Thank you very much for creating this app.  It works great... for 30 seconds.... ha!

Knock on port 1, 2, 3, enables RDP and has the Status as ENABLED. But after 30 seconds, it then is disabled and I am kicked out of the RDP session.

Windows 10 x64 version 21H1

I think this is due to the fDenyTSConnections DWORD key being modified.  A better approach might be to modify the Windows Firewall rules to whitelist the existing connection's IP until disconnected, OR keep RDP enabled until the session has disconnected, then hide it again.

**Comment by asheroto** [comment]
Another note.... you might consider (or even an option) enabling/disabling a firewall rule pertaining to RDP rather than enabling/disabling the Terminal Service itself.  That would make it a true port knocker :-)

**Comment by firepacket** [comment]
I agree. Would it be hard to use WPF to setup a single filter to block the RDP port, then use normal packet sniffing to check for the ports and make sure the IP is the same. Grant access to that IP as long as the connection stays open.

Then you're not messing with RDP functionality at all, just adding/removing a filter with WPF.

It could be a mini port knocking firewall!

**Comment by daleffe** [comment]
Hey guys,

Sorry for the lack of responses, I'm full of work here and I ended up not actively following the project.

I'll test the app on Windows 10 and check the behavior you mentioned. I believe that working with the Windows Firewall isn't the best solution, as there may be situations where it's disabled by user.

Something I thought quickly, would be to make this release/lock time configurable so that you could use it for more than 30 seconds (hahaha), what do you think (until I can work on another more suitable solution)?
